src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -457,9 +457,10 @@ public void OnAbort(Exception abortException, bool canRetry = false)
                         return;
                     }
 
+                    bool isRetriable = _state == StreamState.ExpectingStatus || _state == StreamState.ExpectingHeaders;
```

Comment:
can we add a comment explaining that we can't retry if we've received anything on the stream?